Content.Server/Spawners/Components/GenericSpawnerComponent.cs

```diff
@@ -0,0 +1,41 @@
+using Content.Shared.EntityList;
+using Content.Shared.Random;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.Spawners.Components;
+
+/// <summary>
+/// This component spawns entities from a WeightedRandomEntityPrototype(aka entity table).
+/// </summary>
+[RegisterComponent, EntityCategory("Spawner")]
+[Virtual]
+public partial class GenericSpawnerComponent : Component
+{
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField(customTypeSerializer: typeof(PrototypeIdSerializer<WeightedRandomEntityPrototype>))]
+    public string EntityTable = string.Empty;
+
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField]
+    public List<EntProtoId> GameRules = new();
+
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField]
+    public float Chance { get; set; } = 1.0f;
```

Comment:
Please add summary comments indicating what each field does. For example, that indicating that this is a chance for the roll to fail. Even if the purpose feels obvious